swagger-ui/src/main/scala/com/http4s/rho/swagger/ui/SwaggerUiRoutes.scala

```diff
@@ -0,0 +1,111 @@
+package com.http4s.rho.swagger.ui
+
+import cats.effect.{Blocker, ContextShift, Sync}
+import cats.implicits._
+import com.http4s.rho.swagger.ui.SwaggerUiRoutes.Html
+import org.http4s.headers.`Content-Type`
+import org.http4s.rho.RhoRoutes
+import org.http4s.rho.bits.PathAST.CaptureTail
+import org.http4s.rho.swagger.ui.BuildInfo
+import org.http4s.{EntityEncoder, MediaType, Request, Response, StaticFile}
+
+class SwaggerUiRoutes[F[+ _] : Sync : ContextShift](swaggerUiResourcesPath: String,
+                                                    indexHtml: Html,
+                                                    blocker: Blocker) extends RhoRoutes[F] {
+
+  private implicit val htmlEncoder: EntityEncoder[F, Html] =
+    EntityEncoder.stringEncoder[F]
+      .contramap[Html](html => html.html)
+      .withContentType(`Content-Type`(MediaType.text.html).withCharset(org.http4s.Charset.`UTF-8`))
+
+  GET |>> { req: Request[F] =>
+    PermanentRedirect(req.uri / "index.html")
+  }
+
+  GET / "index.html" |>> { () =>
+    indexHtml
+  }
+
+  GET / CaptureTail |>> { (req: Request[F], path: List[String]) =>
+    fetchResource(swaggerUiResourcesPath + path.mkString("/", "/", ""), req)
+  }
+
+  private def fetchResource(path: String, req: Request[F]): F[Response[F]] =
+    StaticFile.fromResource[F](path, blocker, Some(req)).getOrElseF(NotFound(()).map(_.resp))
+
+}
+
+object SwaggerUiRoutes {
+
+  def apply[F[+ _] : Sync : ContextShift](swaggerUrl: String, blocker: Blocker): SwaggerUiRoutes[F] = {
+    val swaggerUiResourcesPath = s"/META-INF/resources/webjars/swagger-ui/${BuildInfo.swaggerUiVersion}/"
+    val indexHtml = defaultIndexHtml(swaggerUrl)
+    new SwaggerUiRoutes[F](swaggerUiResourcesPath, indexHtml, blocker)
+  }
+
+  case class Html(html: String)
```

Comment:
I generally like like the way a good type alias can impact the type signature and make it more readable, but is this buying us anything here?  Generally we will only define the swagger-ui once, we aren't checking that the indexHtml is valid like what might be done with an opaque alias...

Reply:
The sole purpose of the `Html` is to create a new type for the implicit search of `EntityEncoder`. The indexHtml is a `String`, but it needs a custom encoder (with html content type). 

Of course there are different ways of achieving that goal, for example:
```
  private val htmlEncoder: EntityEncoder[F, String] =
    EntityEncoder.stringEncoder[F]
      .withContentType(`Content-Type`(MediaType.text.html).withCharset(org.http4s.Charset.`UTF-8`))

  GET / "index.html" |>> { () =>
    Ok(indexHtml)(implicitly, htmlEncoder)
  }
```
would be an alternative way of doing it.

Reply:
That makes sense.  I think I would bias this explicit approach as to not burden users with the wrapping... especially for something that is used only once like this.

Reply:
Ok for me.